colander/tests/test_colander.py

```diff
@@ -3405,30 +3487,73 @@ def test_imperative_with_implicit_schema_type(self):
         self.assertEqual(schema.schema_type, colander.Sequence)
         self.assertEqual(schema.children[0], node)
 
-    def test_deserialize_drop(self):
+    def test_deserialize_missing_drop(self):
         import colander
         class MySchema(colander.SequenceSchema):
-            a = colander.SchemaNode(colander.String(), missing=colander.drop)
+            a = colander.SchemaNode(
+                colander.String(),
+                default='abc',
+                missing=colander.drop
+                )
         node = MySchema()
         result = node.deserialize([None])
         self.assertEqual(result, [])
         result = node.deserialize([colander.null])
         self.assertEqual(result, [])
+        result = node.deserialize([''])
+        self.assertEqual(result, [])
 
-    def test_serialize_drop_default(self):
+    def test_deserialize_missing_value(self):
+        import colander
+        class MySchema(colander.SequenceSchema):
+            a = colander.SchemaNode(
+                colander.String(),
+                default=colander.drop,
+                missing='abc'
+                )
+        node = MySchema()
+        result = node.deserialize([None])
+        self.assertEqual(result, ['abc'])
+        result = node.deserialize([colander.null])
+        self.assertEqual(result, ['abc'])
+
+    def test_serialize_default_drop(self):
         import colander
         class MySchema(colander.SequenceSchema):
-            a = colander.SchemaNode(colander.String(), default=colander.drop)
+            a = colander.SchemaNode(
+                colander.String(),
+                default=colander.drop,
+                missing='abc'
+                )
         node = MySchema()
+        result = node.serialize([])
+        self.assertEqual(result, [])
         result = node.serialize([colander.null])
         self.assertEqual(result, [])
 
+    def test_serialize_default_value(self):
+        import colander
+        class MySchema(colander.SequenceSchema):
+            a = colander.SchemaNode(
+                colander.String(),
+                default='abc',
+                missing=colander.drop
+                )
+        node = MySchema()
+        # NOTE: This test fails--but what *should* it do
+        # when a Sequence has a subnode with a default?
+        # result = node.serialize([])
+        # self.assertEqual(result, ['abc'])
```

Comment:
I agree with @tseaver that this should just return an empty list unless there is actually an item in the sequence, such as with `colander.null`.